src/Administration/Resources/app/administration/src/module/sw-flow/component/modals/sw-flow-set-order-state-modal/index.js

```diff
@@ -42,6 +42,7 @@ Component.register('sw-flow-set-order-state-modal', {
 
         stateMachineStateCriteria() {
             const criteria = new Criteria(1, null);
+            criteria.setLimit(50);
```

Comment:
what if there are more than 50? :wink: 

we had such problems in the past. but I don't know anymore an example :grin: maybe @seggewiss or @jleifeld can help here to find a more general solution

Reply:
The solution is to null set the limit of the Criteria as its already done here. Especially for State Machine states this should not be a problem, as there will most likely won't be many of them. So to be more precise : 
`const criteria = new Criteria(1, null);`

Will give you all State Machine states. If it is not suitable to display all at once, a more complex change is required. For example a Select which works with continous scrolling (pagination).